wallets/cosmjs-extension/src/extension/client.ts

```diff
@@ -0,0 +1,170 @@
+import { chainRegistryChainToKeplr } from '@chain-registry/keplr';
+import { StdSignature, StdSignDoc } from '@cosmjs/amino';
+import { Bip39, Random } from '@cosmjs/crypto';
+import {
+  Algo,
+  OfflineDirectSigner,
+  DirectSecp256k1HdWallet,
+} from '@cosmjs/proto-signing';
+
+import {
+  BroadcastMode,
+  ChainRecord,
+  ExtendedHttpEndpoint,
+  SignType,
+  SuggestToken,
+} from '@cosmos-kit/core';
+import { DirectSignDoc, SignOptions, WalletClient } from '@cosmos-kit/core';
+
+import { Cosmjs } from './types';
+import Long from 'long';
+
+export class CosmjsClient implements WalletClient {
+  readonly client: Cosmjs;
+  private _defaultSignOptions: SignOptions = {
+    preferNoSetFee: false,
+    preferNoSetMemo: true,
+    disableBalanceCheck: true,
+  };
+
+  get defaultSignOptions() {
+    return this._defaultSignOptions;
+  }
+
+  setDefaultSignOptions(options: SignOptions) {
+    this._defaultSignOptions = options;
+  }
+
+  constructor(client: Cosmjs) {
+    this.client = client;
+  }
+
+  async enable(chainIds: string | string[]) {
+    await this.client.enable(chainIds);
+  }
+
+  async suggestToken({ chainId, tokens, type }: SuggestToken) {
+    if (type === 'cw20') {
+      for (const { contractAddress } of tokens) {
+        await this.client.suggestCW20Token(chainId, contractAddress);
+      }
+    }
+  }
+
+  async addChain(chainInfo: ChainRecord) {
+    const suggestChain = chainRegistryChainToKeplr(
+      chainInfo.chain,
+      (chainInfo.assetList ? [chainInfo.assetList] : []) as any
+    );
+
+    if (chainInfo.preferredEndpoints?.rest?.[0]) {
+      (suggestChain.rest as string | ExtendedHttpEndpoint) =
+        chainInfo.preferredEndpoints?.rest?.[0];
+    }
+
+    if (chainInfo.preferredEndpoints?.rpc?.[0]) {
+      (suggestChain.rpc as string | ExtendedHttpEndpoint) =
+        chainInfo.preferredEndpoints?.rpc?.[0];
+    }
+
+    await this.client.experimentalSuggestChain(suggestChain);
+  }
+
+  async disconnect() {
+    await this.client.disconnect();
+  }
+
+  async getSimpleAccount(chainId: string) {
+    const { wallet, address, username } = await this.getAccount(chainId);
+    return {
+      namespace: 'cosmos',
+      chainId,
+      wallet,
+      address,
+      username,
+    };
+  }
+
+  generateMnemonic(): string {
+    return Bip39.encode(Random.getBytes(16)).toString();
+  }
+
+  async getAccount(chainId: string) {
+    // const key = await this.client.getKey(chainId);
+    const mnemonic = this.generateMnemonic();
```

Comment:
this seems that it would be non-deterministic — shouldn't the client somehow have a reference to the mnemonic? and we create it in the test itself?